app/controllers/movies_controller.rb

```diff
@@ -3,44 +3,39 @@
 class MoviesController < ApplicationController
   # GET /movies
   def index
-    movies = Movie.all
-
-    render json: movies
+    movies = Movie.where('title LIKE ?', "%#{movie_params[:title]}%")
+    render json: movies, status: :ok
   end
 
   # GET /movies/:id
   def show
-    @movie = set_movie
-
-    render json: @movie
+    render json: movie, status: :ok
   end
 
   # POST /movies
   def create
     movie = Movie.create(movie_params)
 
-    if @movie.valid?
-      render json: @movie, status: :created
+    if movie.save
```

Comment:
"create" already save it to db, so you should stay with `movie.valid?`
https://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-create